x-pack/platform/plugins/shared/stack_connectors/server/connector_types/cases_webhook/service.ts

```diff
@@ -87,10 +91,12 @@ export const createExternalService = (
     throw Error(`[Action]${i18n.NAME}: Wrong configuration.`);
   }
 
+  const mergedHeaders = mergeConfigHeadersWithSecretHeaders(headers, secrets.secretHeaders);
+
   const headersWithBasicAuth = combineHeadersWithBasicAuthHeader({
     username: basicAuth.auth?.username,
     password: basicAuth.auth?.password,
-    headers,
+    headers: mergedHeaders,
   });
 
   const axiosInstance = axios.create({
```

Comment:
Not related to this PR: Why do stack connectors use axios and not the core http service?